docs/dev/sql-nested-function-where-clause.md

```diff
@@ -0,0 +1,413 @@
+
+
+### 1.3 Example Data and Usage
+
+A basic example from data mapping to response from SQL plugin. This example is very basic but useful for understanding the `nested` functions use.
+
+**Mapping:**
+```json
+{
+  "mappings": {
+    "properties": {
+      "message": {
+        "type": "nested",
+        "properties": {
+          "info": {
+            "type": "keyword",
+            "index": "true"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+**Dataset:**
+```json
+{"index":{"_id":"1"}}
+{"message":{"info":"a"}}
+{"index":{"_id":"2"}}
+{"message":{"info":"c"}}
+```
+
+**Query:**
+- `SELECT * FROM nested_objects WHERE nested(message, message.info = 'a');`
+- `SELECT * FROM nested_objects WHERE nested(message.info) = 'a';`
+
+Both queries produce same response from OpenSearch.
+
+**Response:**
+```json
+{
+  "schema": [
+    {
+      "name": "message",
+      "type": "nested"
+    }
+  ],
+  "datarows": [
+    [
+      [
+        {
+          "info": "a"
+        }
+      ]
+    ]
+  ],
+  "total": 1,
+  "size": 1,
+  "status": 200
+}
+```
+
+
+### 1.4 Example Queries
+
+A basic nested function in the `SELECT` clause. This example filters the `nested` object `message` and the inner field `info` with the value of `a`.
+- `SELECT * FROM nested_objects WHERE nested(message.info) = 'a' OR nested(message.author) = 'elm';`
+
+Using the same query as above but with syntax option 1, the response from OpenSearch will be the same, but the DSL used in the query will differ.
+- `SELECT * FROM nested_objects WHERE nested(message, message.info = 'a' OR messate.author = 'elm');`
+
+This example has a nested function in the `SELECT` clause and the `WHERE` clause of the query. Using the `nested` function in the `SELECT` clause will flatten the response nested objects from OpenSearch.
+- `SELECT nested(message.info) FROM nested_objects WHERE nested(message.info) = 'a';`
+
+When using syntax option 1 we need separate function calls when the `path` is not identical. This query exemplifies how a user can use multiple queries with differing `path` values. 
+- `SELECT * FROM nested_objects WHERE nested(message, message.info = 'a') OR nested(comment, comment.data = '123');`
+
+
+## 2 Architecture Diagrams
+### 2.1 Sequence Diagram for filter Push Down
+A sequence diagram illustrating the execution path for `FilterQueryBuilder` push down. Depending on which syntax option is used for the `nested` function, different DSL will be formed in the output query. The following examples in sections [2.3]() and [2.4]() overview how these different queries are formed through the SQL plugin push down semantics. Query execution times have nominal difference and achieve the same goal of filtering documents based on a nested fields literal value. Differing object trees will be pushed through the filter push down sequence to form an output DSL query.
+
+```mermaid
+sequenceDiagram
```

Comment:
TBH I don't know what value this sequence diagram adds to the design.  There isn't anything specific here to nested.